drivers/flash/flash_stm32.c

```diff
@@ -431,6 +432,16 @@ flash_stm32_get_parameters(const struct device *dev)
 	return &flash_stm32_parameters;
 }
 
+/* Gives the total logical device size in bytes and return 0. */
+static int flash_stm32_get_size(const struct device *dev, uint64_t *size)
+{
+	ARG_UNUSED(dev);
+
+	*size = (uint64_t)(LL_GetFlashSize() * 1024);
```

Comment:
I guess we're not dealing with Terabyte flashes yet. That said, i could be better to cast `LL_GetFlashSize()` return value before the multiplication. 
By the way, I think `1024` should be an explicit unsigned value. 
Suggestion:
```c
	*size = (uint64_t)LL_GetFlashSize() * 1024U;